shared/reducers/chat.js

```diff
@@ -110,21 +110,20 @@ function reducer (state: State = initialState, action: Actions) {
         .set('inbox', newInboxStates)
     }
     case Constants.pendingMessageWasSent: {
-      const {outboxID, messageID, messageState} = action.payload
+      const {conversationIDKey, message, messageState} = action.payload
       const newConversationStates = state.get('conversationStates').update(
-        action.payload.conversationIDKey,
+        conversationIDKey,
         initialConversation,
         conversation => {
-          const index = conversation.get('messages').findIndex(item => item.outboxID === outboxID)
+          const index = conversation.get('messages').findIndex(item => item.outboxID === message.outboxID)
           if (index < 0) {
             console.warn("Couldn't find an outbox entry to modify")
             return conversation
           }
           return conversation.updateIn(['messages', index], item => ({
-            ...item,
-            messageID,
```

Comment:
The receiver still uses the outboxID, it just gets it from message.outboxID now, instead of having it be in the payload directly.  Does that make sense?